packages/analytics-js-plugins/src/deviceModeDestinations/utils.ts

```diff
@@ -22,65 +45,126 @@ const isIntegrationSDKEvaluated = (pluginName: string, modName: string, logger?:
   }
 };
 
-const pause = (time: number) =>
+const wait = (time: number) =>
+  // eslint-disable-next-line compat/compat
   new Promise(resolve => {
-    window.setTimeout(resolve, time);
+    globalThis.setTimeout(resolve, time);
   });
 
-const createIntegrationInstance = (
-  modName: string,
-  pluginName: string,
-  intg: ClientIntegration,
-  integrationInstance: any,
+const createDestinationInstance = (
+  destSDKIdentifier: string,
+  sdkTypeName: string,
+  dest: Destination,
+  state: ApplicationState,
+  logger?: ILogger,
 ) => {
-  // TODO: why we pass all analytics instance here? used in browser.constructor of each integration ????
-  // TODO: create the correct values from te state instead of dummy hardcoded ones
-  // TODO: should we just pass the write key and then get the instance from global object?
-  return new integrationInstance[modName](
-    intg.config,
+  const analytics = (globalThis as any).rudderanalytics;
```

Comment:
Lets discuss on this line.

i would expect that this is the instance of Analytics, rudderanalytics[writeKey] to avoid issues when multiple isntances exit